skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import operator
+from functools import wraps
+
+from ._intersection import IntervalTree
+from skbio.util._decorator import experimental
+
+
+class Interval:
+    """Stores the location and metadata of an interval feature.
```

Comment:
If "interval" isn't the right term for describing this data structure (i.e. supports non-contiguous intervals), the class should be renamed from `Interval` to something else. That being said, I really really really want this PR merged, and @ebolyen, @thermokarst, and I can't think of a better name that isn't outright verbose. I'm okay with leaving the current class name and docstring descriptions; we can always rename later if someone comes up with a better name.